src/EventManagement.Domain/OrderLine.cs

```diff
@@ -25,20 +26,45 @@ public class OrderLine
 		public string ProductVariantName { get; set; }
 		public string ProductVariantDescription { get; set; }
 
+		public int? RefundOrderId { get; private set; }
+		public Order RefundOrder { get; private set; }
+		public bool IsRefund => RefundOrderId.HasValue;
+
 		/// <summary>
 		/// A string that uniquely identifies a product-variant combination
 		/// </summary>
-		public string ItemCode => ProductVariantId.HasValue ? $"K{ProductId}-{ProductVariantId}" : $"K{ProductId}";
+		public string ItemCode 
+		{ 
+			get 
+			{
+				if(IsRefund)
+				{
+					return $"R{RefundOrderId}";
+				}
+				return ProductVariantId.HasValue ? $"K{ProductId}-{ProductVariantId}" : $"K{ProductId}";
+			}
+		}
 
 		public decimal Price { get; set; }
 		public decimal VatPercent { get; set; } = 0;
 
 		public string Comments { get; set; }
 
 		// Navigational properties
+		[InverseProperty("OrderLines")]
 		public Order Order { get; set; }
 		public Product Product { get; set; }
 		public ProductVariant ProductVariant { get; set; }
 
+		public static OrderLine CreateRefundOrderLine(Order forOrder)
+		{
+			return new OrderLine
+			{
+				RefundOrderId = forOrder.OrderId,
+				ProductName = $"Refund for Order #{forOrder.OrderId}",
+				Price = -forOrder.TotalAmount
```

Comment:
Do we need quantity here as well?

Reply:
It's taken care of in `Order.TotalAmount`.

https://github.com/gldraphael/EventManagement/blob/1e24ebabd458b3eb28afe181387fcc336cafdfe6/src/EventManagement.Domain/Order.cs#L107-L109